packages/x-data-grid/src/hooks/features/headerFiltering/useGridHeaderFiltering.ts

```diff
@@ -39,15 +39,15 @@ export const useGridHeaderFiltering = (
         return {
           ...state,
           headerFiltering: {
-            enabled: isHeaderFilteringEnabled ?? false,
+            enabled: props.headerFilters ?? false,
```

Comment:
Shouldn't it have a default value? If so, can we remove the `?? false`?

Reply:
It's [initialized](https://github.com/mui/mui-x/blob/master/packages/x-data-grid-pro/src/DataGridPro/useDataGridProComponent.tsx#L105) in the `Pro` package, for MIT use-cases it will be `undefined` so initializing it here.

Reply:
Aren't props initalized with [this](https://github.com/mui/mui-x/blob/6b72f371eedb6ec455e2f338e06f276584305ae5/packages/x-data-grid/src/DataGrid/useDataGridProps.ts#L29)? Would it make sense to do it there?

Btw is the prop going to be visible to community users?

Reply:
> Aren't props initalized with [this](https://github.com/mui/mui-x/blob/6b72f371eedb6ec455e2f338e06f276584305ae5/packages/x-data-grid/src/DataGrid/useDataGridProps.ts#L29)? Would it make sense to do it there?

Yes, but it's done in the package where the feature belongs, for example, `props.headerFilters` belongs to the Pro package and is [defined](https://github.com/mui/mui-x/blob/6b72f371eedb6ec455e2f338e06f276584305ae5/packages/x-data-grid-pro/src/DataGridPro/useDataGridProProps.ts#L34) there too.

Although we could shift the initialization to the MIT, keeping it possible to be `undefined` in MIT would be closer to the actual scenario.

For use-cases with `<DataGrid />`
The value will be `undefined` so `?? false` will come into play

For use cases with `Pro` and `Premium`
The value will already be defined, which will be used.

Wdyt?

> Btw is the prop going to be visible to community users?

No, it's only exported from `/internals` and used in the respective package!

Reply:
> The value will be undefined so ?? false will come into play

And do the typings reflect that?

I had a similar problem with the pinned state & slots, and I felt it was nicer to have everything defined to a good default value, rather than leaving some stuff potentially undefined.

But it's fine if you think it's better the way it is, just wanted to throw the idea.

Reply:
> And do the typings reflect that?

Yes, I [wrapped](https://github.com/mui/mui-x/pull/12365/files#diff-18991c189d0e0b75b030b0889cfbee75905b8b4c19361d0a4b975a1a3ac26945R808-R811) these into `Partial` for the MIT version to reflect that.

Reply:
Ok, sounds good.